pennylane/transforms/measurement_grouping.py

```diff
@@ -51,6 +51,9 @@ def measurement_grouping(tape, obs_list, coeffs_list):
         tapes.append(new_tape)
 
     def processing_fn(res):
-        return qml.math.sum([qml.math.dot(c, r) for c, r in zip(coeffs_groupings, res)])
+        dot_products = [
+            qml.math.dot(qml.math.convert_like(c, r), r) for c, r in zip(coeffs_groupings, res)
+        ]
+        return qml.math.sum(qml.math.stack(dot_products))

```

Comment:
:thinking: Maybe good to leave a comment on the PR for subtle changes like this. Why we need to edit `measurement_grouping`?

Reply:
That's odd, I don't recall making this change. Maybe something weird happened while merging, I'll fix it.

Reply:
Oh no, this was me while I was testing your PR @Lucaman99!

This actually fixed a bug, can we keep this in?

Reply:
There were two issues here:

* the Hamiltonian class casts the `coeffs` to a list, which is a bit annoying. It requires casting it back to a tensor by doing `qml.math.convert_like(c, r), r)`.

* Secondly, the resulting list of dot products needs to be stacked before summation to support TensorFlow, which does not like summing a list of tensors.

Reply:
In the end this transform is so simple! (Once you know what to do!)